Build error on Windows (template build, MSVC and MinGW)

**Operating system or device, Godot version, GPU Model and driver (if graphics related):** Windows 10, Godot Git https://github.com/godotengine/godot/commit/62cb43bb8dec8f5078012b58ff06a2077078565f, MSVC 2017 and MinGW-w64 7.2.0

The build of export templates (`tools=no target=release` for this build) fails, the MinGW error messages are attached below:

```
Compiling ==> modules\gdnative\gd_native_library_editor.cpp
=====
modules\gdnative\gd_native_library_editor.cpp:4:6: error: 'GDNativeLibraryEditor' has not been declared
 void GDNativeLibraryEditor::_find_gdnative_singletons(EditorFileSystemDirectory *p_dir, const Set<String> &enabled_list) {
      ^~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:4:55: error: variable or field '_find_gdnative_singletons' declared void
 void GDNativeLibraryEditor::_find_gdnative_singletons(EditorFileSystemDirectory *p_dir, const Set<String> &enabled_list) {
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:4:55: error: 'EditorFileSystemDirectory' was not declared in this scope
modules\gdnative\gd_native_library_editor.cpp:4:82: error: 'p_dir' was not declared in this scope
 void GDNativeLibraryEditor::_find_gdnative_singletons(EditorFileSystemDirectory *p_dir, const Set<String> &enabled_list) {
                                                                                  ^~~~~
modules\gdnative\gd_native_library_editor.cpp:4:89: error: expected primary-expression before 'const'
 void GDNativeLibraryEditor::_find_gdnative_singletons(EditorFileSystemDirectory *p_dir, const Set<String> &enabled_list) {
                                                                                         ^~~~~
modules\gdnative\gd_native_library_editor.cpp:40:6: error: 'GDNativeLibraryEditor' has not been declared
 void GDNativeLibraryEditor::_update_libraries() {
      ^~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp: In function 'void _update_libraries()':
modules\gdnative\gd_native_library_editor.cpp:42:2: error: 'updating' was not declared in this scope
  updating=true;
  ^~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:43:2: error: 'libraries' was not declared in this scope
  libraries->clear();
  ^~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:47:6: error: 'ProjectSettings' has not been declared
  if (ProjectSettings::get_singleton()->has("gdnative/singletons")) {
      ^~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:48:17: error: 'ProjectSettings' has not been declared
   enabled_paths=ProjectSettings::get_singleton()->get("gdnative/singletons");
                 ^~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:55:2: error: 'EditorFileSystemDirectory' was not declared in this scope
  EditorFileSystemDirectory *fs = EditorFileSystem::get_singleton()->get_filesystem();
  ^~~~~~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:55:29: error: 'fs' was not declared in this scope
  EditorFileSystemDirectory *fs = EditorFileSystem::get_singleton()->get_filesystem();
                             ^~
modules\gdnative\gd_native_library_editor.cpp:55:34: error: 'EditorFileSystem' has not been declared
  EditorFileSystemDirectory *fs = EditorFileSystem::get_singleton()->get_filesystem();
                                  ^~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:57:3: error: '_find_gdnative_singletons' was not declared in this scope
   _find_gdnative_singletons(fs,enabled_list);
   ^~~~~~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:57:3: note: suggested alternative: 'godot_gdnative_init_fn'
   _find_gdnative_singletons(fs,enabled_list);
   ^~~~~~~~~~~~~~~~~~~~~~~~~
   godot_gdnative_init_fn
modules\gdnative\gd_native_library_editor.cpp: At global scope:
modules\gdnative\gd_native_library_editor.cpp:64:6: error: 'GDNativeLibraryEditor' has not been declared
 void GDNativeLibraryEditor::_item_edited() {
      ^~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp: In function 'void _item_edited()':
modules\gdnative\gd_native_library_editor.cpp:65:6: error: 'updating' was not declared in this scope
  if (updating)
      ^~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:68:2: error: 'TreeItem' was not declared in this scope
  TreeItem *item = libraries->get_edited();
  ^~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:68:12: error: 'item' was not declared in this scope
  TreeItem *item = libraries->get_edited();
            ^~~~
modules\gdnative\gd_native_library_editor.cpp:68:12: note: suggested alternative: 'tm'
  TreeItem *item = libraries->get_edited();
            ^~~~
            tm
modules\gdnative\gd_native_library_editor.cpp:68:19: error: 'libraries' was not declared in this scope
  TreeItem *item = libraries->get_edited();
                   ^~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:76:6: error: 'ProjectSettings' has not been declared
  if (ProjectSettings::get_singleton()->has("gdnative/singletons")) {
      ^~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:77:17: error: 'ProjectSettings' has not been declared
   enabled_paths=ProjectSettings::get_singleton()->get("gdnative/singletons");
                 ^~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:89:3: error: 'ProjectSettings' has not been declared
   ProjectSettings::get_singleton()->set("gdnative/singletons",enabled_paths);
   ^~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:91:3: error: 'ProjectSettings' has not been declared
   ProjectSettings::get_singleton()->set("gdnative/singletons",Variant());
   ^~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp: At global scope:
modules\gdnative\gd_native_library_editor.cpp:95:6: error: 'GDNativeLibraryEditor' has not been declared
 void GDNativeLibraryEditor::_notification(int p_what) {
      ^~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp: In function 'void _notification(int)':
modules\gdnative\gd_native_library_editor.cpp:97:14: error: 'NOTIFICATION_VISIBILITY_CHANGED' was not declared in this scope
  if (p_what==NOTIFICATION_VISIBILITY_CHANGED) {
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:98:7: error: 'is_visible_in_tree' was not declared in this scope
   if (is_visible_in_tree()) {
       ^~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:98:7: note: suggested alternative: '__builtin_free'
   if (is_visible_in_tree()) {
       ^~~~~~~~~~~~~~~~~~
       __builtin_free
modules\gdnative\gd_native_library_editor.cpp: At global scope:
modules\gdnative\gd_native_library_editor.cpp:104:6: error: 'GDNativeLibraryEditor' has not been declared
 void GDNativeLibraryEditor::_bind_methods() {
      ^~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp: In function 'void _bind_methods()':
modules\gdnative\gd_native_library_editor.cpp:106:49: error: 'GDNativeLibraryEditor' has not been declared
  ClassDB::bind_method(D_METHOD("_item_edited"),&GDNativeLibraryEditor::_item_edited);
                                                 ^~~~~~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp: At global scope:
modules\gdnative\gd_native_library_editor.cpp:109:1: error: 'GDNativeLibraryEditor' does not name a type; did you mean 'GDNativeLibrary'?
 GDNativeLibraryEditor::GDNativeLibraryEditor()
 ^~~~~~~~~~~~~~~~~~~~~
 GDNativeLibrary

=====
scons: *** [modules\gdnative\gd_native_library_editor.windows.opt.64.o] Error 1
scons: building terminated because of errors.
```

This may also be relevant, here's the HTML5 build failing (in WebAssembly mode):

```
Compiling ==> modules\gdnative\gd_native_library_editor.cpp
modules\gdnative\gd_native_library_editor.cpp:4:6: error: use of undeclared identifier 'GDNativeLibraryEditor'; did you mean 'GDNativeLibrary'?
void GDNativeLibraryEditor::_find_gdnative_singletons(EditorFileSystemDirectory *p_dir, const Set<String> &enabled_list) {
     ^~~~~~~~~~~~~~~~~~~~~
     GDNativeLibrary
modules\gdnative/gdnative.h:40:7: note: 'GDNativeLibrary' declared here
class GDNativeLibrary : public Resource {
      ^
modules\gdnative\gd_native_library_editor.cpp:4:55: error: unknown type name 'EditorFileSystemDirectory'
void GDNativeLibraryEditor::_find_gdnative_singletons(EditorFileSystemDirectory *p_dir, const Set<String> &enabled_list) {
                                                      ^
modules\gdnative\gd_native_library_editor.cpp:19:4: error: unknown type name 'TreeItem'
                        TreeItem *ti = libraries->create_item(libraries->get_root());
                        ^
modules\gdnative\gd_native_library_editor.cpp:19:19: error: use of undeclared identifier 'libraries'
                        TreeItem *ti = libraries->create_item(libraries->get_root());
                                       ^
modules\gdnative\gd_native_library_editor.cpp:19:42: error: use of undeclared identifier 'libraries'
                        TreeItem *ti = libraries->create_item(libraries->get_root());
                                                              ^
modules\gdnative\gd_native_library_editor.cpp:23:24: error: use of undeclared identifier 'TreeItem'
                        ti->set_cell_mode(1,TreeItem::CELL_MODE_RANGE);
                                            ^
modules\gdnative\gd_native_library_editor.cpp:40:6: error: use of undeclared identifier 'GDNativeLibraryEditor'; did you mean 'GDNativeLibrary'?
void GDNativeLibraryEditor::_update_libraries() {
     ^~~~~~~~~~~~~~~~~~~~~
     GDNativeLibrary
modules\gdnative/gdnative.h:40:7: note: 'GDNativeLibrary' declared here
class GDNativeLibrary : public Resource {
      ^
modules\gdnative\gd_native_library_editor.cpp:40:29: error: out-of-line definition of '_update_libraries' does not match any declaration in 'GDNativeLibrary'
void GDNativeLibraryEditor::_update_libraries() {
                            ^~~~~~~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:42:2: error: use of undeclared identifier 'updating'
        updating=true;
        ^
modules\gdnative\gd_native_library_editor.cpp:43:2: error: use of undeclared identifier 'libraries'
        libraries->clear();
        ^
modules\gdnative\gd_native_library_editor.cpp:44:2: error: use of undeclared identifier 'libraries'
        libraries->create_item(); //rppt
        ^
modules\gdnative\gd_native_library_editor.cpp:47:6: error: use of undeclared identifier 'ProjectSettings'
        if (ProjectSettings::get_singleton()->has("gdnative/singletons")) {
            ^
modules\gdnative\gd_native_library_editor.cpp:48:17: error: use of undeclared identifier 'ProjectSettings'
                enabled_paths=ProjectSettings::get_singleton()->get("gdnative/singletons");
                              ^
modules\gdnative\gd_native_library_editor.cpp:55:2: error: unknown type name 'EditorFileSystemDirectory'
        EditorFileSystemDirectory *fs = EditorFileSystem::get_singleton()->get_filesystem();
        ^
modules\gdnative\gd_native_library_editor.cpp:55:34: error: use of undeclared identifier 'EditorFileSystem'
        EditorFileSystemDirectory *fs = EditorFileSystem::get_singleton()->get_filesystem();
                                        ^
modules\gdnative\gd_native_library_editor.cpp:60:2: error: use of undeclared identifier 'updating'
        updating=false;
        ^
modules\gdnative\gd_native_library_editor.cpp:64:6: error: use of undeclared identifier 'GDNativeLibraryEditor'; did you mean 'GDNativeLibrary'?
void GDNativeLibraryEditor::_item_edited() {
     ^~~~~~~~~~~~~~~~~~~~~
     GDNativeLibrary
modules\gdnative/gdnative.h:40:7: note: 'GDNativeLibrary' declared here
class GDNativeLibrary : public Resource {
      ^
modules\gdnative\gd_native_library_editor.cpp:64:29: error: out-of-line definition of '_item_edited' does not match any declaration in 'GDNativeLibrary'
void GDNativeLibraryEditor::_item_edited() {
                            ^~~~~~~~~~~~
modules\gdnative\gd_native_library_editor.cpp:65:6: error: use of undeclared identifier 'updating'
        if (updating)
            ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
scons: *** [modules\gdnative\gd_native_library_editor.javascript.opt.webassembly.bc] Error 1
scons: building terminated because of errors.
```

--- a/modules/gdnative/gd_native_library_editor.cpp
+++ b/modules/gdnative/gd_native_library_editor.cpp
@@ -1,3 +1,4 @@
+#ifdef TOOLS_ENABLED
 #include "gd_native_library_editor.h"
 #include "gdnative.h"
 
@@ -118,3 +119,4 @@ GDNativeLibraryEditor::GDNativeLibraryEditor()
 	updating=false;
 	libraries->connect("item_edited",this,"_item_edited");
 }
+#endif

